src/Tools/IiifUrlValidator.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: dpino
+ * Date: 11/27/18
+ * Time: 1:06 PM
+ */
+
+namespace Drupal\format_strawberryfield\Tools;
+
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\ServerException;
+use GuzzleHttp\Exception\ConnectException;
+
+class IiifUrlValidator {
+
+  use StringTranslationTrait;
+
+  /**
+   * @var \GuzzleHttp\Client;
+   */
+  protected $httpClient;
+
+  public function __construct () {
+    $this->httpClient = $this->httpClient = \Drupal::httpClient();
+  }
+
+  public function checkPublicUrl($publicUrl) {
+    // check if there's a value entered
+    if (trim($publicUrl)) {
+      $isLocalhost = strpos($publicUrl, 'localhost');
```

Comment:
Ok, strpos will just find if there is a localhost.. but it could be also 127.0.0.1! Sorry, this is hard. I would parse the URL into its constituent (Search in php.net, is super easy) just take the domain and check if its in array ('localhost', 127.0.0.1', '0.0.0.0')